File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque


def bfs(n, computers, computer, visited):
    visited[computer] = 1
    que = deque([computer])
    
    while que:
        now = que.popleft()
        visited[now] = True
        for next in range(n):
            if next != now:
                if computers[now][next] == 1:
                    if visited[next] == False:
                        que.append(next)


def solution(n, computers):
    answer = 0
    visited = [0] * n
    for computer in range(n):
        if not visited[computer]:
            bfs(n, computers, computer, visited)
            answer += 1
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    
    var visited = new Array(n).fill(false);
    
    const bfs = (idx) => {
        var que = [idx];
        visited[idx] = true;
        
        while (que.length) {
            const now = que.splice(0, 1);
            for (let next in computers[now]) {
                if (computers[now][next] && !visited[next]) {
                    visited[next] = true;
                    que.push(next);
                }
            }
        }
    }
    
    for (let i = 0; i < n; i++) {
        if (!visited[i]) {
            bfs(i);
            answer ++;
        }
    }
    
    return answer;
}